src/components/Filters/Filters.tsx

```diff
@@ -63,16 +70,47 @@ function Filters({ partners, setFilteredPartners }: Props) {
     return Array.from(uniqueLanguages).sort((a, b) => a.localeCompare(b))
   }, [partners])
 
+  const getUrlFilters = () => {
+    const filters = { ...EMPTY_FILTER }
+    const { query } = router
+    for (const key of Object.keys(query)) {
+      const filterKey = key as keyof Filters
+      if (Object.values(FilterType).includes(filterKey)) {
+        const value = query[key]
+        filters[filterKey] = typeof value === 'string' ? [value] : [...(value || [])]
+      }
+    }
+
+    return filters
+  }
+
   useEffect(() => {
+    const checkBoxStates: CheckBoxStates = {}
     for (const category of dropdownContent) {
       for (const item of Object.values(category.options)) {
-        setCheckBoxState((prev) => ({ ...prev, [`${category.key}#${item}`]: false }))
+        checkBoxStates[getCheckboxKey(category.key, item)] = false
```

Comment:
I don't think it is a good idea to override the useState object... I'll investigate this a little bit more 🤔

Reply:
Oh sorry, my bad! I am not directly modifying the useState object, it is an object with a name similar to that of the state. I've already renamed it so it's not confusing 🙌🏼